File: 1403.非递增顺序的最小子序列.go
/*
 * @lc app=leetcode.cn id=1403 lang=golang
 *
 * [1403] 非递增顺序的最小子序列
 */
package leetcode
import "sort"
// @lc code=start
func minSubsequence(nums []int) (res []int) {
	sum, cur := 0, 0
	for i := 0; i < len(nums); i++ {
		sum += nums[i]
	}
	sort.Slice(nums, func (i, j int) bool {return nums[i] > nums[j]})
	for i := 0; i < len(nums); i++ {
		cur += nums[i]
		res = append(res, nums[i])
		if cur * 2 > sum {
			return 
		}
	}
	return
}
// @lc code=end

File: 1403.非递增顺序的最小子序列.cpp
/*
 * @lc app=leetcode.cn id=1403 lang=cpp
 *
 * [1403] 非递增顺序的最小子序列
 */
#include <vector>
#include <algorithm>
using namespace std;
// @lc code=start
class Solution {
public:
    vector<int> minSubsequence(vector<int>& nums) {
        int sum = 0;
        for (auto v : nums) {
            sum += v;
        }
        sort(nums.begin(), nums.end());
        int cur = 0;
        vector<int> res;
        for (auto iter = nums.end() - 1; iter != nums.end(); iter--) {
            cur += *iter;
            res.emplace_back(*iter);
            if (cur * 2 > sum) 
                return res;
        }
        return res;
    }
};
// @lc code=end

